diff --git a/src/3rdparty/assimp/src/contrib/unzip/ioapi.h b/src/3rdparty/assimp/src/contrib/unzip/ioapi.h
index 63ec641..03b2c29 100644
--- a/src/3rdparty/assimp/src/contrib/unzip/ioapi.h
+++ b/src/3rdparty/assimp/src/contrib/unzip/ioapi.h
@@ -31,6 +31,11 @@
 #endif
 #endif
 
+#ifdef _Z_OF
+#undef OF
+#define OF _Z_OF
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
diff --git a/src/3rdparty/assimp/src/contrib/unzip/unzip.h b/src/3rdparty/assimp/src/contrib/unzip/unzip.h
index b247937..1008494 100644
--- a/src/3rdparty/assimp/src/contrib/unzip/unzip.h
+++ b/src/3rdparty/assimp/src/contrib/unzip/unzip.h
@@ -66,6 +66,10 @@ typedef unzFile__ *unzFile;
 typedef voidp unzFile;
 #endif
 
+#ifdef _Z_OF
+#undef OF
+#define OF _Z_OF
+#endif
 
 #define UNZ_OK                          (0)
 #define UNZ_END_OF_LIST_OF_FILE         (-100)
