From 82017437e2a858adefaa41ade0e6140017c1d2e2 Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Tue, 7 Apr 2020 19:01:08 +0300
Subject: [PATCH 3/3] Workaround for logout confirmation message box issue

---
 src/dolphinmainwindow.cpp | 7 +++++++
 src/dolphinmainwindow.h   | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/src/dolphinmainwindow.cpp b/src/dolphinmainwindow.cpp
index cfb48edb2..c25340f51 100644
--- a/src/dolphinmainwindow.cpp
+++ b/src/dolphinmainwindow.cpp
@@ -538,6 +538,12 @@ void DolphinMainWindow::showEvent(QShowEvent* event)
 
 void DolphinMainWindow::closeEvent(QCloseEvent* event)
 {
+    // Workaround for Qt >= 5.14.0: if close event was already processed and accepted, skip processing it again
+    if (m_closeEventAccepted) {
+        KXmlGuiWindow::closeEvent(event);
+        return;
+    }
+
     // Find out if Dolphin is closed directly by the user or
     // by the session manager because the session is closed
     bool closedByUser = true;
@@ -654,6 +660,7 @@ void DolphinMainWindow::closeEvent(QCloseEvent* event)
 
     GeneralSettings::setVersion(CurrentDolphinVersion);
     GeneralSettings::self()->save();
+    m_closeEventAccepted = true;
 
     KXmlGuiWindow::closeEvent(event);
 }
diff --git a/src/dolphinmainwindow.h b/src/dolphinmainwindow.h
index 3af940fd9..e4eccfc7b 100644
--- a/src/dolphinmainwindow.h
+++ b/src/dolphinmainwindow.h
@@ -711,6 +711,8 @@ private:
     QMenu m_searchTools;
     KFileItemActions m_fileItemActions;
 
+    bool m_closeEventAccepted = false;
+
     friend class DolphinMainWindowTest;
 };
 
-- 
2.39.2

