From bb24431836b34044a635603c24bf3e6f5b439b5c Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Tue, 7 Apr 2020 19:01:08 +0300
Subject: [PATCH 3/3] Workaround for logout confirmation message box issue

---
 src/dolphinmainwindow.cpp | 7 +++++++
 src/dolphinmainwindow.h   | 1 +
 2 files changed, 8 insertions(+)

diff --git a/src/dolphinmainwindow.cpp b/src/dolphinmainwindow.cpp
index 2ab50d6f6..ab9594957 100644
--- a/src/dolphinmainwindow.cpp
+++ b/src/dolphinmainwindow.cpp
@@ -512,6 +512,12 @@ void DolphinMainWindow::showEvent(QShowEvent* event)
 
 void DolphinMainWindow::closeEvent(QCloseEvent* event)
 {
+    // Workaround for Qt >= 5.14.0: if close event was already processed and accepted, skip processing it again
+    if (m_closeEventAccepted) {
+        KXmlGuiWindow::closeEvent(event);
+        return;
+    }
+
     // Find out if Dolphin is closed directly by the user or
     // by the session manager because the session is closed
     bool closedByUser = true;
@@ -628,6 +634,7 @@ void DolphinMainWindow::closeEvent(QCloseEvent* event)
 
     GeneralSettings::setVersion(CurrentDolphinVersion);
     GeneralSettings::self()->save();
+    m_closeEventAccepted = true;
 
     KXmlGuiWindow::closeEvent(event);
 }
diff --git a/src/dolphinmainwindow.h b/src/dolphinmainwindow.h
index 46515cc8b..5c7ca0e2a 100644
--- a/src/dolphinmainwindow.h
+++ b/src/dolphinmainwindow.h
@@ -671,6 +671,7 @@ private:
     QMenu m_searchTools;
     KFileItemActions m_fileItemActions;
 
+    bool m_closeEventAccepted = false;
 };
 
 inline DolphinViewContainer* DolphinMainWindow::activeViewContainer() const
-- 
2.32.0

