From 77fab3e064b710ab2fb4eca04516166a4a81591c Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Tue, 7 Apr 2020 19:01:08 +0300
Subject: [PATCH 3/3] Workaround for logout confirmation message box issue

---
 src/dolphinmainwindow.cpp | 7 +++++++
 src/dolphinmainwindow.h   | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/src/dolphinmainwindow.cpp b/src/dolphinmainwindow.cpp
index b6dde6f4b..56f07bfbe 100644
--- a/src/dolphinmainwindow.cpp
+++ b/src/dolphinmainwindow.cpp
@@ -539,6 +539,12 @@ void DolphinMainWindow::showEvent(QShowEvent *event)
 
 void DolphinMainWindow::closeEvent(QCloseEvent *event)
 {
+    // Workaround for Qt >= 5.14.0: if close event was already processed and accepted, skip processing it again
+    if (m_closeEventAccepted) {
+        KXmlGuiWindow::closeEvent(event);
+        return;
+    }
+
     // Find out if Dolphin is closed directly by the user or
     // by the session manager because the session is closed
     bool closedByUser = true;
@@ -652,6 +658,7 @@ void DolphinMainWindow::closeEvent(QCloseEvent *event)
 
     GeneralSettings::setVersion(CurrentDolphinVersion);
     GeneralSettings::self()->save();
+    m_closeEventAccepted = true;
 
     KXmlGuiWindow::closeEvent(event);
 }
diff --git a/src/dolphinmainwindow.h b/src/dolphinmainwindow.h
index 00cb1c2e9..2a585a360 100644
--- a/src/dolphinmainwindow.h
+++ b/src/dolphinmainwindow.h
@@ -712,6 +712,8 @@ private:
     QMenu m_searchTools;
     KFileItemActions m_fileItemActions;
 
+    bool m_closeEventAccepted = false;
+
     friend class DolphinMainWindowTest;
 };
 
-- 
2.41.0

