From c9ec2723596a20da0dabc9d6b5a57b9e2cfb4bb4 Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Tue, 7 Apr 2020 19:01:08 +0300
Subject: [PATCH 3/3] Workaround for logout confirmation message box issue

---
 src/dolphinmainwindow.cpp | 7 +++++++
 src/dolphinmainwindow.h   | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/src/dolphinmainwindow.cpp b/src/dolphinmainwindow.cpp
index d2f8ce513..8972c17ba 100644
--- a/src/dolphinmainwindow.cpp
+++ b/src/dolphinmainwindow.cpp
@@ -632,6 +632,12 @@ void DolphinMainWindow::showEvent(QShowEvent *event)
 
 void DolphinMainWindow::closeEvent(QCloseEvent *event)
 {
+    // Workaround for Qt >= 5.14.0: if close event was already processed and accepted, skip processing it again
+    if (m_closeEventAccepted) {
+        KXmlGuiWindow::closeEvent(event);
+        return;
+    }
+
     // Find out if Dolphin is closed directly by the user or
     // by the session manager because the session is closed
     bool closedByUser = true;
@@ -749,6 +755,7 @@ void DolphinMainWindow::closeEvent(QCloseEvent *event)
 
     GeneralSettings::setVersion(CurrentDolphinVersion);
     GeneralSettings::self()->save();
+    m_closeEventAccepted = true;
 
     KXmlGuiWindow::closeEvent(event);
 }
diff --git a/src/dolphinmainwindow.h b/src/dolphinmainwindow.h
index c5ae02926..35b93d58c 100644
--- a/src/dolphinmainwindow.h
+++ b/src/dolphinmainwindow.h
@@ -798,6 +798,8 @@ private:
 
     KRecentFilesAction *m_recentFiles = nullptr;
 
+    bool m_closeEventAccepted = false;
+
     friend class DolphinMainWindowTest;
 };
 
-- 
2.51.2

