From 51e0a500c37ad8eeddbfd276685f0e4a0ce0aeb3 Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Fri, 12 Jun 2020 21:44:05 +0300
Subject: [PATCH] Fix warnings

---
 dbusaddons/fcitxqtconnection.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/dbusaddons/fcitxqtconnection.cpp b/dbusaddons/fcitxqtconnection.cpp
index 4770c4c..817d911 100644
--- a/dbusaddons/fcitxqtconnection.cpp
+++ b/dbusaddons/fcitxqtconnection.cpp
@@ -119,8 +119,14 @@ void FcitxQtConnectionPrivate::initialize() {
 
 void FcitxQtConnectionPrivate::finalize() {
     m_serviceWatcher->removeWatchedService(m_serviceName);
-    m_watcher->removePaths(m_watcher->files());
-    m_watcher->removePaths(m_watcher->directories());
+    if (!m_watcher->files().isEmpty())
+    {
+        m_watcher->removePaths(m_watcher->files());
+    }
+    if (!m_watcher->directories().isEmpty())
+    {
+        m_watcher->removePaths(m_watcher->directories());
+    }
     disconnect(m_watcher, &QFileSystemWatcher::fileChanged, this,
                &FcitxQtConnectionPrivate::socketFileChanged);
     disconnect(m_watcher, &QFileSystemWatcher::directoryChanged, this,
-- 
2.26.2

