diff --git a/intern/libmv/CMakeLists.txt b/intern/libmv/CMakeLists.txt
index 6fcf664..201eaac 100644
--- a/intern/libmv/CMakeLists.txt
+++ b/intern/libmv/CMakeLists.txt
@@ -49,12 +49,12 @@ if(WITH_LIBMV)
   add_definitions(-DLIBMV_GFLAGS_NAMESPACE=${GFLAGS_NAMESPACE})
 
   list(APPEND INC
-    ${GFLAGS_INCLUDE_DIRS}
-    ${GLOG_INCLUDE_DIRS}
     ../guardedalloc
   )
 
   list(APPEND INC_SYS
+    ${GFLAGS_INCLUDE_DIRS}
+    ${GLOG_INCLUDE_DIRS}
     ${EIGEN3_INCLUDE_DIRS}
     ${PNG_INCLUDE_DIRS}
     ${ZLIB_INCLUDE_DIRS}
