diff --git a/source/creator/CMakeLists.txt b/source/creator/CMakeLists.txt
index d9f8c80..bc97226 100644
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -387,12 +387,6 @@ endif()
 # Install Targets (Generic, All Platforms)
 
 
-# Important to make a clean install each time, else old scripts get loaded.
-install(
-  CODE
-  "file(REMOVE_RECURSE ${TARGETDIR_VER})"
-)
-
 if(WITH_PYTHON)
   # install(CODE "message(\"copying blender scripts...\")")
 
