--- modules/system-source/system-source.c.back	2017-03-06 20:17:58.569512050 +0300
+++ modules/system-source/system-source.c	2017-03-06 20:19:49.503518553 +0300
@@ -222,22 +222,24 @@
 _is_running_in_linux_container(void)
 {
   FILE *f;
-  char line[2048];
+  char *line = NULL;
+  size_t len = 0;
   gboolean container = FALSE;
 
-  f = fopen("/proc/1/cgroup", "r");
+  f = fopen("/proc/1/environ", "r");
   if (!f)
     return FALSE;
 
-  while (fgets(line, sizeof(line), f) != NULL)
+  while (getdelim(&line, &len, '\0', f) > 0)
     {
-      if (line[strlen(line) - 2] != '/')
+      if (strcmp(line, "container=lxc") == 0)
         {
           container = TRUE;
           break;
         }
     }
 
+  free(line);
   fclose (f);
 
   return container;
