From 8aef848c4e2903ef234904c4bdefaaa73297acdc Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Sat, 23 Mar 2019 14:46:50 +0300
Subject: [PATCH] Fix -Wreturn-type warning for libjpeg check

---
 lib/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 169df3c..499de1c 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -18,7 +18,7 @@ file(WRITE ${JPEGLIB_VERSION_CHECK_PATH} "
 #include <stdlib.h>
 #include <jpeglib.h>
 
-int main(void) { printf(\"%d\\\n\", JPEG_LIB_VERSION); }
+int main(void) { printf(\"%d\\\n\", JPEG_LIB_VERSION); return 0; }
 ")
 
 try_run(JPEGLIB_RUN_RESULT JPEGLIB_COMPILE_RESULT
-- 
2.19.2

