From f6bd4776b45a9021903bde7cb5eb2eb90e32feaa Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Tue, 10 Oct 2017 20:50:50 +0300
Subject: [PATCH] qtbase: Always allow to make examples.

---
 mkspecs/features/qt_example_installs.prf | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/mkspecs/features/qt_example_installs.prf b/mkspecs/features/qt_example_installs.prf
index 72b47bce27..b3637f0c7b 100644
--- a/mkspecs/features/qt_example_installs.prf
+++ b/mkspecs/features/qt_example_installs.prf
@@ -30,13 +30,6 @@ probase = $$relative_path($$_PRO_FILE_PWD_, $$moduleRoot/examples)
 isEmpty(probase)|contains(probase, ^\\..*): \
     return()
 
-isEmpty(_QMAKE_CACHE_) {
-    moduleRootRelativeToBuildDir = $$relative_path($$moduleRoot, $$OUT_PWD)
-    # Check if OUT_PWD is inside module root
-    equals(moduleRootRelativeToBuildDir, .)|contains(moduleRootRelativeToBuildDir, \(\.\./\)+\(\.\.\)?): \
-        error("You cannot build examples inside the Qt source tree, except as part of a proper Qt build.")
-}
-
 contains(TEMPLATE, "vc.*"): \
     return()
 
-- 
2.24.1

