From 6729177a7514ec1867ddb539049ff19913ce32c6 Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Wed, 3 Jul 2019 20:18:39 +0300
Subject: [PATCH] Fix compilation on 32bit

---
 source/encoder/motion.cpp | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/source/encoder/motion.cpp b/source/encoder/motion.cpp
index f0e3aaa..4e18ef2 100644
--- a/source/encoder/motion.cpp
+++ b/source/encoder/motion.cpp
@@ -1061,19 +1061,19 @@ me_hex2:
                 size_t dy = (size_t)i * stride;
 #define SADS(k, x0, y0, x1, y1, x2, y2, x3, y3) \
     sad_x4(fenc, \
-           fref_base x0 * i + (y0 - 2 * k + 4) * dy, \
-           fref_base x1 * i + (y1 - 2 * k + 4) * dy, \
-           fref_base x2 * i + (y2 - 2 * k + 4) * dy, \
-           fref_base x3 * i + (y3 - 2 * k + 4) * dy, \
-           stride, costs + 4 * k); \
+           fref_base + (x0) * i + ((y0) - 2 * (k) + 4) * dy, \
+           fref_base + (x1) * i + ((y1) - 2 * (k) + 4) * dy, \
+           fref_base + (x2) * i + ((y2) - 2 * (k) + 4) * dy, \
+           fref_base + (x3) * i + ((y3) - 2 * (k) + 4) * dy, \
+           stride, costs + 4 * (k)); \
     fref_base += 2 * dy;
 #define ADD_MVCOST(k, x, y) costs[k] += p_cost_omvx[x * 4 * i] + p_cost_omvy[y * 4 * i]
 #define MIN_MV(k, dx, dy)     if ((omv.y + (dy) >= mvmin.y) & (omv.y + (dy) <= mvmax.y)) { COPY2_IF_LT(bcost, costs[k], dir, dx * 16 + (dy & 15)) }
 
-                SADS(0, +0, -4, +0, +4, -2, -3, +2, -3);
-                SADS(1, -4, -2, +4, -2, -4, -1, +4, -1);
-                SADS(2, -4, +0, +4, +0, -4, +1, +4, +1);
-                SADS(3, -4, +2, +4, +2, -2, +3, +2, +3);
+                SADS(0,  0, -4,  0,  4, -2, -3,  2, -3);
+                SADS(1, -4, -2,  4, -2, -4, -1,  4, -1);
+                SADS(2, -4,  0,  4,  0, -4,  1,  4,  1);
+                SADS(3, -4,  2,  4,  2, -2,  3,  2,  3);
                 ADD_MVCOST(0, 0, -4);
                 ADD_MVCOST(1, 0, 4);
                 ADD_MVCOST(2, -2, -3);
-- 
2.21.0

