diff --git a/unix/ibus/ibus_config.cc b/unix/ibus/ibus_config.cc
index 2ae02f1..08b2f0a 100644
--- a/unix/ibus/ibus_config.cc
+++ b/unix/ibus/ibus_config.cc
@@ -205,7 +205,7 @@ bool IbusConfig::IsActiveOnLaunch() const {
 
   // The default value is off as IBus team's recommendation.
   // https://github.com/google/mozc/issues/201
-  return false;
+  return true;
 }
 
 bool IbusConfig::IsMozcRendererEnabled() const {
diff --git a/unix/ibus/ibus_config.textproto b/unix/ibus/ibus_config.textproto
index e4829dd..647b3a2 100644
--- a/unix/ibus/ibus_config.textproto
+++ b/unix/ibus/ibus_config.textproto
@@ -66,7 +66,7 @@ engines {
   symbol : "A"
   composition_mode : DIRECT
 }
-active_on_launch: False
+active_on_launch: True
 mozc_renderer {
   # Set 'False' to use IBus' candidate window.
   enabled : True
