diff --git a/src/unix/ibus/gen_mozc_xml.py b/src/unix/ibus/gen_mozc_xml.py
index 87e6aec0..fa614cf5 100755
--- a/src/unix/ibus/gen_mozc_xml.py
+++ b/src/unix/ibus/gen_mozc_xml.py
@@ -113,7 +113,7 @@ def GetIbusConfigTextProto(engines):
     for key, value in engine.items():
       output.append(GetTextProtoElement(key, value))
     output.append('}')
-  output.append('active_on_launch: False')
+  output.append('active_on_launch: True')
   output.append('mozc_renderer {')
   output.append("  # Set 'False' to use IBus' candidate window.")
   output.append('  enabled : True')
diff --git a/src/unix/ibus/ibus_config.cc b/src/unix/ibus/ibus_config.cc
index a1fb0361..e72559a9 100644
--- a/src/unix/ibus/ibus_config.cc
+++ b/src/unix/ibus/ibus_config.cc
@@ -185,7 +185,7 @@ bool IbusConfig::IsActiveOnLaunch() const {
 
   // The default value is off as IBus team's recommentation.
   // https://github.com/google/mozc/issues/201
-  return false;
+  return true;
 }
 
 bool IbusConfig::IsMozcRendererEnabled() const {
