From d5532128d8d3fa7c150de095690ad74632a3caa2 Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Sun, 28 Jun 2020 12:03:52 +0300
Subject: [PATCH] Change default view

---
 src/qml/main.qml | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/qml/main.qml b/src/qml/main.qml
index 4ada6dc..c3595f0 100644
--- a/src/qml/main.qml
+++ b/src/qml/main.qml
@@ -155,11 +155,7 @@ ApplicationWindow {
         console.log("Orientation", Screen.orientation)
 
         //Initial view
-        if (!Settings.hasAccessToken) {
-            topbar.setCurrentIndex(5)
-        } else {
-            topbar.setCurrentIndex(1)
-        }
+        topbar.setCurrentIndex(2)
 
         //Check version
         if (Settings.versionCheckEnabled) {
-- 
2.26.2

