From be22282a0f897b0bf6615fe9d0a17cd71002eead Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Sun, 28 Jun 2020 18:33:59 +0300
Subject: [PATCH] Hide headers only in player view

---
 src/qml/PlayerView.qml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/qml/PlayerView.qml b/src/qml/PlayerView.qml
index 949aa85..e0b9707 100644
--- a/src/qml/PlayerView.qml
+++ b/src/qml/PlayerView.qml
@@ -619,6 +619,7 @@ Page {
                 if (!root.headersVisible || headerBarArea.containsMouse || bottomBarArea.containsMouse) return
 
                 if (renderer.status === "PAUSED" || renderer.status === "STOPPED") return
+                if (!view.isItemInView(playerView)) return;
 
                 // Bug?: MouseArea doesn't work over Controls
                 var controls = [ favBtn, chatBtn, playBtn, resetBtn, volumeBtn, volumeSlider, seekBar, sourcesBox, cropBtn, fsBtn];
-- 
2.26.2

