From 52d404616f34b6b92d7eaf8472f8bc7416df410e Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Sun, 3 Jan 2021 11:40:29 +0300
Subject: [PATCH] Fix documentation install location

---
 cmake/QtDocsHelpers.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/QtDocsHelpers.cmake b/cmake/QtDocsHelpers.cmake
index 8e4bd28918..fb8ffca9a5 100644
--- a/cmake/QtDocsHelpers.cmake
+++ b/cmake/QtDocsHelpers.cmake
@@ -146,7 +146,7 @@ function(qt_internal_add_docs)
 
     # generate .qch
     set(qch_file_name ${doc_target}.qch)
-    set(qch_file_path ${qdoc_output_dir}/${qch_file_name})
+    set(qch_file_path ${index_dir}/${qch_file_name})
 
     foreach(target_prefix qch_top_level_docs qch_repo_docs qch_docs)
         add_custom_target(${target_prefix}_${target}
@@ -176,7 +176,7 @@ function(qt_internal_add_docs)
         )
 
         install(FILES "${qch_file_path}"
-                DESTINATION "${INSTALL_DOCDIR}/${qch_file_name}"
+                DESTINATION "${INSTALL_DOCDIR}"
                 COMPONENT _install_qch_docs_${target}
                 EXCLUDE_FROM_ALL
         )
-- 
2.26.2

