diff --git a/src/corelib/global/qsimd_p.h b/src/corelib/global/qsimd_p.h
index 1feedbbb..22cc6a1d 100644
--- a/src/corelib/global/qsimd_p.h
+++ b/src/corelib/global/qsimd_p.h
@@ -213,11 +213,10 @@ asm(
 // ELF loader also supports a "haswell/" subdir (e.g., /usr/lib/haswell).
 #  define ARCH_HASWELL_MACROS       (__AVX2__ + __BMI__ + __BMI2__ + __F16C__ + __FMA__ + __LZCNT__)
 #  if ARCH_HASWELL_MACROS != 0
-#    if ARCH_HASWELL_MACROS != 6
-#      error "Please enable all x86-64-v3 extensions; you probably want to use -march=haswell or -march=x86-64-v3 instead of -mavx2"
-#    endif
+#    if ARCH_HASWELL_MACROS == 6
 static_assert(ARCH_HASWELL_MACROS, "Undeclared identifiers indicate which features are missing.");
-#    define __haswell__       1
+#      define __haswell__       1
+#    endif
 #  endif
 #  undef ARCH_HASWELL_MACROS
 
@@ -227,11 +226,10 @@ static_assert(ARCH_HASWELL_MACROS, "Undeclared identifiers indicate which featur
 // 6th generation (codename "Skylake").
 #  define ARCH_SKX_MACROS           (__AVX512F__ + __AVX512BW__ + __AVX512CD__ + __AVX512DQ__ + __AVX512VL__)
 #  if ARCH_SKX_MACROS != 0
-#    if ARCH_SKX_MACROS != 5
-#      error "Please enable all x86-64-v4 extensions; you probably want to use -march=skylake-avx512 or -march=x86-64-v4 instead of -mavx512f"
-#    endif
+#    if ARCH_SKX_MACROS == 5
 static_assert(ARCH_SKX_MACROS, "Undeclared identifiers indicate which features are missing.");
-#    define __skylake_avx512__  1
+#      define __skylake_avx512__  1
+#    endif
 #  endif
 #  undef ARCH_SKX_MACROS
 #endif  /* Q_PROCESSOR_X86 */
