diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d0ffccb6..10f8d3a7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -290,8 +290,8 @@ if(BUILD_TESTS)
   target_link_libraries(conky conky_core ${conky_libs})
   install(TARGETS conky_core
           RUNTIME DESTINATION bin
-          LIBRARY DESTINATION lib
-          ARCHIVE DESTINATION lib)
+          LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+          ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 else()
   add_executable(conky main.cc ${conky_sources} ${optional_sources})
   target_link_libraries(conky ${conky_libs})
@@ -300,7 +300,14 @@ endif()
 # Install libtcp-portmon too?
 install(TARGETS conky
         RUNTIME DESTINATION bin
-        LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib)
+        LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+        ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
+
+if(BUILD_PORT_MONITORS)
+  install(TARGETS tcp-portmon
+          RUNTIME DESTINATION bin
+          LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+          ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
+endif(BUILD_PORT_MONITORS)
 
 print_target_properties(conky)
