diff --git a/cr3qt/CMakeLists.txt b/cr3qt/CMakeLists.txt
index 9830ddcd..67e46ad7 100644
--- a/cr3qt/CMakeLists.txt
+++ b/cr3qt/CMakeLists.txt
@@ -214,8 +214,7 @@ INSTALL( FILES ${CR3_CHANGELOG} DESTINATION share/doc/cr3 )
 INSTALL( FILES ${QM_FILES} DESTINATION share/cr3/i18n )
 INSTALL( FILES src/desktop/cr3.desktop DESTINATION share/applications )
 INSTALL( FILES src/desktop/cr3.appdata.xml DESTINATION share/metainfo )
-INSTALL( FILES src/desktop/cr3.png DESTINATION share/pixmaps )
-INSTALL( FILES src/desktop/cr3.xpm DESTINATION share/pixmaps )
+INSTALL( FILES ../icons_src/logo/source/cr3icon.png DESTINATION share/pixmaps RENAME cr3.png)
 
 ELSE()
 
diff --git a/cr3wx/CMakeLists.txt b/cr3wx/CMakeLists.txt
index d6421b22..f8f22474 100644
--- a/cr3wx/CMakeLists.txt
+++ b/cr3wx/CMakeLists.txt
@@ -22,8 +22,7 @@ IF (UNIX)
   INSTALL( DIRECTORY ../cr3qt/data/skins DESTINATION share/cr3/skins )
   INSTALL( FILES ../cr3qt/src/desktop/cr3.desktop DESTINATION share/applications )
   INSTALL( FILES ../cr3qt/src/desktop/cr3.appdata.xml DESTINATION share/metainfo )
-  INSTALL( FILES ../cr3qt/src/desktop/cr3.png DESTINATION share/pixmaps )
-  INSTALL( FILES ../cr3qt/src/desktop/cr3.xpm DESTINATION share/pixmaps )
+  INSTALL( FILES ../icons_src/logo/source/cr3icon.png DESTINATION share/pixmaps RENAME cr3.png)
 ELSE()
   INSTALL( TARGETS cr3 RUNTIME DESTINATION . )
   INSTALL( DIRECTORY ../cr3qt/data/ DESTINATION .
