diff --git a/src/kwalletsettings.kcfg b/src/kwalletsettings.kcfg
index e0caf5b5..3ac32046 100644
--- a/src/kwalletsettings.kcfg
+++ b/src/kwalletsettings.kcfg
@@ -20,7 +20,7 @@
             <default>true</default>
         </entry>
         <entry name="KWalletDEnabled" key="Enabled" type="Bool">
-            <default>true</default>
+            <default>false</default>
         </entry>
         <entry key="Idle Timeout" type="Int">
             <default>10</default>
