diff --git a/apps/cmstapp/code/control_box/controlbox.cpp b/apps/cmstapp/code/control_box/controlbox.cpp
index 76a58d4..71d34a5 100755
--- a/apps/cmstapp/code/control_box/controlbox.cpp
+++ b/apps/cmstapp/code/control_box/controlbox.cpp
@@ -2529,6 +2529,7 @@ void ControlBox::assembleTrayIcon()
    QPixmap pxm = prelimicon.pixmap(prelimicon.actualSize((QSize(22,22) *= iconscale)) );
    QImage src = pxm.toImage();
    QImage dest = QImage(src.width(), src.height(), QImage::Format_ARGB32);
+   dest.setDevicePixelRatio(src.devicePixelRatio());
    QPainter painter(&dest);
    if (trayiconbackground.isValid() && src.hasAlphaChannel() ) {
       painter.setCompositionMode(QPainter::CompositionMode_Source);
