From 9e7eab511eb2a9f78bf6b8ab82d0b6b13dad81ad Mon Sep 17 00:00:00 2001
From: Matteo Bruni <mbruni@codeweavers.com>
Date: Fri, 22 Feb 2019 21:45:52 +0100
Subject: [PATCH] wined3d: Workaround for Nvidia glGenerateMipmap() bug.

See bug 46531.
---
 dlls/wined3d/view.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dlls/wined3d/view.c b/dlls/wined3d/view.c
index dc67ea9cbd9..e8a05c88fe5 100644
--- a/dlls/wined3d/view.c
+++ b/dlls/wined3d/view.c
@@ -905,6 +905,7 @@ void shader_resource_view_generate_mipmaps(struct wined3d_shader_resource_view *
     struct wined3d_texture_gl *texture_gl;
     struct wined3d_context *context;
     struct gl_texture *gl_tex;
+    GLint draw_fbo;
     DWORD location;
     BOOL srgb;
 
@@ -923,6 +924,9 @@ void shader_resource_view_generate_mipmaps(struct wined3d_shader_resource_view *
     for (i = 0; i < layer_count; ++i)
         wined3d_texture_load_location(&texture_gl->t, i * level_count + base_level, context, location);
 
+    gl_info->gl_ops.gl.p_glGetIntegerv(GL_DRAW_FRAMEBUFFER_BINDING, &draw_fbo);
+    gl_info->fbo_ops.glBindFramebuffer(GL_DRAW_FRAMEBUFFER, 0);
+
     if (view_gl->gl_view.name)
     {
         shader_resource_view_gl_bind_and_dirtify(view_gl, context);
@@ -947,6 +951,8 @@ void shader_resource_view_generate_mipmaps(struct wined3d_shader_resource_view *
     gl_info->fbo_ops.glGenerateMipmap(texture_gl->target);
     checkGLcall("glGenerateMipMap()");
 
+    gl_info->fbo_ops.glBindFramebuffer(GL_DRAW_FRAMEBUFFER, draw_fbo);
+
     for (i = 0; i < layer_count; ++i)
     {
         for (j = base_level + 1; j <= max_level; ++j)
-- 
2.19.2

