diff --git a/modules/module-mpris.c b/modules/module-mpris.c
index 10ee815..c271069 100644
--- a/modules/module-mpris.c
+++ b/modules/module-mpris.c
@@ -128,7 +128,7 @@ static void players_unref (Players *players)
     return;
 
   g_mutex_clear (&players->lock);
-  g_clear_object (&players->items);
+  g_hash_table_destroy (players->items);
   g_clear_object (&players->conn);
   g_clear_object (&players->cancellable);
   g_free (players);
