diff --git a/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp b/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp
index 21e8779f..612f8e31 100644
--- a/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp
+++ b/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp
@@ -232,6 +232,8 @@ void UISelectorWindow::sltShowSelectorWindowContextMenu(const QPoint &position)
             m_pToolBar->show();
         else
             m_pToolBar->hide();
+
+        gEDataManager->setSelectorWindowToolBarVisible(pResult->isChecked());
     }
     else if (pResult == pShowToolBarText)
     {
@@ -239,6 +241,8 @@ void UISelectorWindow::sltShowSelectorWindowContextMenu(const QPoint &position)
                                        ? Qt::ToolButtonTextUnderIcon : Qt::ToolButtonIconOnly);
         m_pToolbarTools->setToolButtonStyle(pResult->isChecked()
                                             ? Qt::ToolButtonTextUnderIcon : Qt::ToolButtonIconOnly);
+
+        gEDataManager->setSelectorWindowToolBarTextVisible(pResult->isChecked());
     }
     else if (pResult == pShowStatusBar)
     {
@@ -246,6 +250,8 @@ void UISelectorWindow::sltShowSelectorWindowContextMenu(const QPoint &position)
             statusBar()->show();
         else
             statusBar()->hide();
+
+        gEDataManager->setSelectorWindowStatusBarVisible(pResult->isChecked());
     }
 }
 
@@ -2157,13 +2163,6 @@ void UISelectorWindow::saveSettings()
         gEDataManager->setSelectorWindowToolsOrderGlobal(m_pToolbarTools->tabOrderGlobal());
     }
 
-    /* Save toolbar and statusbar visibility: */
-    {
-        gEDataManager->setSelectorWindowToolBarVisible(!m_pToolBar->isHidden());
-        gEDataManager->setSelectorWindowToolBarTextVisible(m_pToolBar->toolButtonStyle() == Qt::ToolButtonTextUnderIcon);
-        gEDataManager->setSelectorWindowStatusBarVisible(!statusBar()->isHidden());
-    }
-
     /* Save splitter handle position: */
     {
         gEDataManager->setSelectorWindowSplitterHints(m_pSplitter->sizes());
