diff --git a/src/3rdparty/chromium/base/bind_internal.h b/src/3rdparty/chromium/base/bind_internal.h
index d59300dfd..85fcc50a5 100644
--- a/src/3rdparty/chromium/base/bind_internal.h
+++ b/src/3rdparty/chromium/base/bind_internal.h
@@ -733,6 +733,7 @@ bool QueryCancellationTraitsImpl(BindStateBase::CancellationQueryMode mode,
           functor, std::get<indices>(bound_args)...);
   }
   NOTREACHED();
+  return false;
 }
 
 // Relays |base| to corresponding CallbackCancellationTraits<>::Run(). Returns
diff --git a/src/3rdparty/chromium/ui/base/clipboard/clipboard.h b/src/3rdparty/chromium/ui/base/clipboard/clipboard.h
index 6403ff002..24588bb18 100644
--- a/src/3rdparty/chromium/ui/base/clipboard/clipboard.h
+++ b/src/3rdparty/chromium/ui/base/clipboard/clipboard.h
@@ -50,6 +50,7 @@ class COMPONENT_EXPORT(BASE_CLIPBOARD) Clipboard : public base::ThreadChecker {
         return false;
     }
     NOTREACHED();
+    return false;
   }
 
   // Sets the list of threads that are allowed to access the clipboard.
diff --git a/src/core/web_contents_delegate_qt.cpp b/src/core/web_contents_delegate_qt.cpp
index 38f139513..1440e26d4 100644
--- a/src/core/web_contents_delegate_qt.cpp
+++ b/src/core/web_contents_delegate_qt.cpp
@@ -100,6 +100,7 @@ static WebContentsAdapterClient::JavaScriptConsoleMessageLevel mapToJavascriptCo
     case blink::mojom::ConsoleMessageLevel::kError:
         return WebContentsAdapterClient::Error;
     }
+    return WebContentsAdapterClient::Info;
 }
 
 WebContentsDelegateQt::WebContentsDelegateQt(content::WebContents *webContents, WebContentsAdapterClient *adapterClient)
