diff --git a/scripts/xdg-open.in b/scripts/xdg-open.in
index 9817c5f..7252222 100644
--- a/scripts/xdg-open.in
+++ b/scripts/xdg-open.in
@@ -435,7 +435,13 @@ open_lxde()
             file="$(pwd)/$file"
         fi
 
-        pcmanfm "$file"
+        if pcmanfm --help 2>/dev/null; then
+          pcmanfm "$file"
+        elif pcmanfm-qt --help 2>/dev/null; then
+          pcmanfm-qt "$file"
+        else
+          open_generic "$1"
+        fi
     else
         open_generic "$1"
     fi
