diff --git a/miscsplashutils-0.1.8/fbtruetype/Makefile b/miscsplashutils-0.1.8/fbtruetype/Makefile
index 37d14eb..89833ba 100644
--- a/miscsplashutils-0.1.8/fbtruetype/Makefile
+++ b/miscsplashutils-0.1.8/fbtruetype/Makefile
@@ -19,16 +19,14 @@ LIB      ?= lib
 
 LIBDIR   = $(PREFIX)/$(LIB)
 
-PROGRAMS = fbtruetype fbtruetype.static
+PROGRAMS = fbtruetype
 OBJECTS  = fbtruetype.o messages.o console.o ttf.o luxisri.o
 
 LDFLAGS  += -L$(LIBDIR)
-LIBSS    = $(shell pkg-config freetype2 --libs --static) -lm
 LIBSD    = $(shell pkg-config freetype2 --libs) -lm
 
 CFLAGS   += -I/usr/include/freetype2
 COMPILE  = $(CC) $(CFLAGS)
-LINKS    = $(CC) $(CFLAGS) $(LDFLAGS) -static -o $@
 LINKD    = $(CC) $(CFLAGS) $(LDFLAGS) -o $@
 
 INSTALL_PROG = install
@@ -47,20 +45,16 @@ all: $(PROGRAMS) strip
 mostlyclean:
 	-rm -f *.o core
 clean:	mostlyclean
-	-rm -f fbtruetype fbtruetype.static
+	-rm -f fbtruetype
 
 fbtruetype: $(OBJECTS)
 	@rm -f fbtruetype
 	$(LINKD) $(LDFLAGS) $(OBJECTS) $(LIBSD)
-fbtruetype.static: $(OBJECTS)
-	@rm -f fbtruetype.static
-	$(LINKS) $(LDFLAGS) $(OBJECTS) $(LIBSS)
 
 install:
 	$(INSTALL_PROG) -D fbtruetype $(DESTDIR)/sbin/fbtruetype
-	$(INSTALL_PROG) -D fbtruetype.static $(DESTDIR)/sbin/fbtruetype.static
 
 strip: $(PROGRAMS)
 	$(STRIP) -s $(PROGRAMS)
 
-.PHONY: mostlyclean clean fbtruetype fbtruetype.static all strip
+.PHONY: mostlyclean clean fbtruetype all strip
