diff --git a/splashutils-1.5.4.4/Makefile.am b/splashutils-1.5.4.4/Makefile.am
index 5a86fb0..767e650 100644
--- a/splashutils-1.5.4.4/Makefile.am
+++ b/splashutils-1.5.4.4/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = libs src docs misc scripts
+SUBDIRS = src docs misc scripts
 ACLOCAL_AMFLAGS = -I m4
 
 include common.mk
diff --git a/splashutils-1.5.4.4/Makefile.in b/splashutils-1.5.4.4/Makefile.in
index e0b168a..3fa505f 100644
--- a/splashutils-1.5.4.4/Makefile.in
+++ b/splashutils-1.5.4.4/Makefile.in
@@ -253,7 +253,7 @@ themedir = @themedir@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = libs src docs misc scripts
+SUBDIRS = src docs misc scripts
 ACLOCAL_AMFLAGS = -I m4
 LIBFREETYPE2_SOURCE_INTERNAL = libs/freetype-2.3.5
 LIBJPEG_SOURCE_INTERNAL = libs/jpeg-6b
@@ -313,7 +313,7 @@ config.h: stamp-h1
 stamp-h1: $(srcdir)/config.h.in $(top_builddir)/config.status
 	@rm -f stamp-h1
 	cd $(top_builddir) && $(SHELL) ./config.status config.h
-$(srcdir)/config.h.in:  $(am__configure_deps) 
+$(srcdir)/config.h.in:  $(am__configure_deps)
 	($(am__cd) $(top_srcdir) && $(AUTOHEADER))
 	rm -f stamp-h1
 	touch $@
diff --git a/splashutils-1.5.4.4/src/Makefile.am b/splashutils-1.5.4.4/src/Makefile.am
index d6ce315..b558c35 100644
--- a/splashutils-1.5.4.4/src/Makefile.am
+++ b/splashutils-1.5.4.4/src/Makefile.am
@@ -242,19 +242,19 @@ fbcondecor_helper_SOURCES = \
 
 fbcondecor_helper_CPPFLAGS  = -DWITH_ERRLIST -DTARGET_KERNEL -DTT_CONFIG_OPTION_BYTECODE_INTERPRETER
 fbcondecor_helper_CFLAGS    = -Os -w -ffunction-sections -fdata-sections -I.@am__isrc@
-fbcondecor_helper_LDFLAGS   = 
-fbcondecor_helper_LDADD     = 
+fbcondecor_helper_LDFLAGS   =
+fbcondecor_helper_LDADD     =
 
-fbcondecor_helper_CFLAGS   += -I$(abs_top_builddir)/libs/libjpeg -I$(LIBJPEG_SOURCE)
-fbcondecor_helper_LDADD    += $(top_builddir)/libs/libjpeg.a
+fbcondecor_helper_CFLAGS   += -I$(includedir)
+fbcondecor_helper_LDFLAGS  += -ljpeg
 if CONFIG_PNG
-fbcondecor_helper_CFLAGS   += -I$(LIBPNG_SOURCE)
-fbcondecor_helper_LDADD    += $(top_builddir)/libs/libpng.a $(top_builddir)/libs/libz.a
+fbcondecor_helper_CFLAGS   += $(shell pkg-config libpng --cflags --static)
+fbcondecor_helper_LDFLAGS  += $(shell pkg-config libpng --libs --static) $(shell pkg-config zlib --libs --static)
 endif
 if CONFIG_TTF_KERNEL
 fbcondecor_helper_SOURCES  += ttf.c ttf.h fbsplash.h
-fbcondecor_helper_CFLAGS   += -I$(LIBFREETYPE2_SOURCE)/include -I$(LIBZ_SOURCE)
-fbcondecor_helper_LDADD    += $(top_builddir)/libs/libfreetype.a
+fbcondecor_helper_CFLAGS   += $(shell pkg-config freetype2 --cflags --static) $(shell pkg-config zlib --cflags --static)
+fbcondecor_helper_LDFLAGS  += $(shell pkg-config freetype2 --libs --static)
 endif
 if CONFIG_KLIBC_SHARED
 fbcondecor_helper_LDFLAGS  += -shared
@@ -266,7 +266,7 @@ endif
 fbcondecor_helper: $(fbcondecor_helper_LDADD) $(fbcondecor_helper_OBJECTS) $(lib_LTLIBRARIES)
 	@$(call infmsg,LD,$@)
 	$(Q)$(MKDIR_P) $(@D)
-	$(Q)$(KLCC) $(fbcondecor_helper_CFLAGS) $(fbcondecor_helper_LDFLAGS) $(fbcondecor_helper_OBJECTS) $(fbcondecor_helper_LDADD) -o $@
+	$(Q)$(CC) $(fbcondecor_helper_CFLAGS) $(fbcondecor_helper_OBJECTS) $(fbcondecor_helper_LDFLAGS) $(fbcondecor_helper_LDADD) -o $@
 	$(Q)$(STRIP) $@
 
 # Override automake's generation of target rules for building fbcondecor_helper object files.
@@ -284,7 +284,7 @@ fbcondecor_helper-ttf.o:
 fbcondecor_helper-%.o: %.c
 	@$(call infmsg,CC,$@)
 	$(Q)$(MKDIR_P) $(@D)
-	$(Q)$(KLCC) $(fbcondecor_helper_CPPFLAGS) $(fbcondecor_helper_CFLAGS) $(INCLUDES) -c $< -o $@
+	$(Q)$(CC) $(fbcondecor_helper_CPPFLAGS) $(fbcondecor_helper_CFLAGS) $(INCLUDES) -c $< -o $@
 else
 fbcondecor_helper_SOURCES = \
 	kernel.c \
-- 
2.0.5

