diff --git a/splashutils-1.5.4.4/scripts/splash_geninitramfs.in b/splashutils-1.5.4.4/scripts/splash_geninitramfs.in
index d50441f..c283f10 100644
--- a/splashutils-1.5.4.4/scripts/splash_geninitramfs.in
+++ b/splashutils-1.5.4.4/scripts/splash_geninitramfs.in
@@ -16,6 +16,50 @@ cleanup()
 	rm -rf ${workdir}
 }

+copy()
+{
+	local from=$1
+	local where=$2
+	local rootdir=$3
+
+	local parentdir="$(dirname "${where}")"
+	mkdir -p "${parentdir}"
+
+	cp -a "$from" "$where"
+	if [ -L "$from" ] ; then
+		local link="$(readlink $from)"
+		if [[ "$link" != /* ]] ; then
+			link="$(dirname $from)/$link"
+		fi
+
+		copy "$link" "${rootdir}/$link" "$rootdir"
+	elif [ -f "$from" ] ; then
+		lddtree -l "$from" | tail -n +2 | while read lib ; do
+			if [ ! -e "${rootdir}/$lib" ] ; then
+				[ -e "${rootdir}/$(dirname $lib)" ] || mkdir -p "${rootdir}/$(dirname $lib)"
+				copy "$lib" "${rootdir}/$lib" "${rootdir}"
+			fi
+		done
+	fi
+}
+
+make_executable()
+{
+	local what=$1
+
+	chmod 755 "$(readlink -e $what)"
+}
+
+copy_executable()
+{
+	local from=$1
+	local where=$2
+	local rootdir=$3
+
+	copy "$from" "$where" "$rootdir"
+	make_executable "$where"
+}
+
 usage()
 {
 	cat <<EOTB
@@ -166,7 +210,7 @@ if [ ! -e ${splash_hlp} ]; then
 fi

 printv "o Copying ${splash_hlp}.."
-cp "${splash_hlp}" "${imgdir}@sbindir@"
+copy_executable "${splash_hlp}" "${imgdir}${splash_hlp}" "${imgdir}"
 ln -sf "${splash_hlp}" "${imgdir}@sbindir@/splash_helper"

 res=${res//,/ }
diff --git a/splashutils-1.5.4.4/src/Makefile.am b/splashutils-1.5.4.4/src/Makefile.am
index 5cf79b4..5ac6618 100644
--- a/splashutils-1.5.4.4/src/Makefile.am
+++ b/splashutils-1.5.4.4/src/Makefile.am
@@ -189,7 +189,7 @@ eexecsbin_PROGRAMS += fbsplashctl
 fbsplashctl_SOURCES         = $(fbsplashd_SOURCES) $(splash_util_SOURCES) fbsplashctl.c
 fbsplashctl_CFLAGS          = $(AM_CFLAGS) $(PTHREAD_STATIC_CFLAGS) $(RT_CFLAGS) $(libfbsplashrender_la_CFLAGS) -DUNIFIED_BUILD
 fbsplashctl_CPPFLAGS        = $(fbsplashd_CPPFLAGS) $(splash_util_CPPFLAGS)
-fbsplashctl_LDFLAGS         = $(AM_LDFLAGS) -all-static
+fbsplashctl_LDFLAGS         = $(AM_LDFLAGS)
 fbsplashctl_LDADD           = libfbsplashrender.la libfbsplash.la $(PTHREAD_STATIC_LIBS) $(RT_LIBS) $(GPM_LIBS)

 if CONFIG_FBCON_DECOR
