From 653bb99639d3dde5094fcfe9cedb1da612b4e703 Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Tue, 7 Sep 2021 21:43:26 +0300
Subject: [PATCH] Keep .desktop suffix when renaming desktop links

---
 kioslave/desktop/kio_desktop.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/kioslave/desktop/kio_desktop.cpp b/kioslave/desktop/kio_desktop.cpp
index 8e4f672cd..e59e283bf 100644
--- a/kioslave/desktop/kio_desktop.cpp
+++ b/kioslave/desktop/kio_desktop.cpp
@@ -196,7 +196,8 @@ void DesktopProtocol::rename(const QUrl &_src, const QUrl &_dest, KIO::JobFlags
 
     QUrl dest;
     rewriteUrl(_dest, dest);
-    const QString destPath = dest.toLocalFile();
+    QString destPath = dest.toLocalFile();
+    QUrl reported_dest = _dest;
 
     if (KDesktopFile::isDesktopFile(srcPath)) {
         QString friendlyName;
@@ -206,6 +207,8 @@ void DesktopProtocol::rename(const QUrl &_src, const QUrl &_dest, KIO::JobFlags
             friendlyName = KIO::decodeFileName(fileName.left(fileName.length() - 8));
         } else {
             friendlyName = KIO::decodeFileName(dest.fileName());
+            destPath.append(QLatin1String(".desktop"));
+            reported_dest.setPath(reported_dest.path().append(QLatin1String(".desktop")));
         }
 
         // Update the value of the Name field in the file.
@@ -217,7 +220,7 @@ void DesktopProtocol::rename(const QUrl &_src, const QUrl &_dest, KIO::JobFlags
     }
 
     if (QFile(srcPath).rename(destPath)) {
-        org::kde::KDirNotify::emitFileRenamedWithLocalPath(_src, _dest, destPath);
+        org::kde::KDirNotify::emitFileRenamedWithLocalPath(_src, reported_dest, destPath);
         finished();
     } else {
         error(KIO::ERR_CANNOT_RENAME, srcPath);
-- 
2.32.0

