diff --git a/src/client/qwaylandinputdevice.cpp b/src/client/qwaylandinputdevice.cpp
index 320e340..72f3415 100644
--- a/src/client/qwaylandinputdevice.cpp
+++ b/src/client/qwaylandinputdevice.cpp
@@ -1005,6 +1005,8 @@ bool QWaylandInputDevice::Pointer::FrameData::hasPixelDelta() const
     case axis_source_continuous:
         return !delta.isNull();
     }
+
+    return false;
 }
 
 QPoint QWaylandInputDevice::Pointer::FrameData::pixelDeltaAndError(QPointF *accumulatedError) const
