--- rawstudio-2.0/plugins/lensfun/lensfun.c.back	2017-01-14 22:31:07.909563325 +0300
+++ rawstudio-2.0/plugins/lensfun/lensfun.c	2017-01-14 22:32:45.345568946 +0300
@@ -652,7 +652,7 @@
 			}
 
 			/* Start threads to apply phase 2, Vignetting and CA Correction */
-			if (effective_flags & (LF_MODIFY_VIGNETTING | LF_MODIFY_CCI)) 
+			if (effective_flags & LF_MODIFY_VIGNETTING) 
 			{
 				/* Phase 2 is corrected inplace, so copy input first */
 				guint y_offset, y_per_thread, threaded_h;
